pkg/gooq/types.go

```diff
@@ -27,10 +28,12 @@ type DatabaseConstraint struct {
 }
 
 type DBInterface interface {
-	Exec(query string, args ...interface{}) (sql.Result, error)
-	Prepare(query string) (*sql.Stmt, error)
-	Queryx(query string, args ...interface{}) (*sqlx.Rows, error)
-	QueryRowx(query string, args ...interface{}) *sqlx.Row
+	sqlx.Execer
+	sqlx.ExecerContext
+	sqlx.Preparer
+	sqlx.PreparerContext
+	sqlx.Queryer
+	sqlx.QueryerContext
```

Comment:
just using the interface provided by sqlx instead of copying the function signature over